File: algorithms/medium/find_the_original_array_of_prefix_xor.cpp
class Solution {
public:
    vector<int> findArray(vector<int>& pref) {
      int acc = 0;
      vector<int> ans;
      for (auto x : pref) {
        ans.push_back(acc ^ x);
        acc = x;
      }
      return ans;
    }
};

File: algorithms/medium/find_the_original_array_of_prefix_xor.ex
defmodule Solution do
  use Bitwise

  @spec find_array(pref :: [integer]) :: [integer]
  def find_array(pref), do: find_array(pref, 0, [])

  def find_array([], _acc, arr), do: arr |> Enum.reverse()

  def find_array([x | xs], acc, arr) do
    find_array(xs, x, [bxor(acc, x) | arr])
  end
end
